staging/src/k8s.io/client-go/discovery/fake/discovery.go

```diff
@@ -31,8 +31,13 @@ import (
 	"k8s.io/client-go/testing"
 )
 
+// FakeDiscovery implements discovery.DiscoveryInterface and sometimes calls testing.Fake.Invoke with an action,
+// but doesn't respect the return value if any. There is a way to fake static values like ServerVersion and the
+// RESTClient using the Faked... fields on the struct.
 type FakeDiscovery struct {
 	*testing.Fake
+	FakedServerVersion *version.Info
+	FakedRESTClient restclient.Interface
```

Comment:
I didn't notice this one in the comment in your diff.  Can we avoid it?

Reply:
Before this; `RESTClient()` always returned `nil`. With this, the user can set this field if they want to make `client.Discovery().RESTClient()` for a faked clientset to work, but it will still default to `nil`.
WDYT?